src/views/Workflow.vue

```diff
@@ -136,20 +142,20 @@ export default {
         QUERIES[queryName].replace('WORKFLOW_ID', workflowId)
       )
       view.subscriptionId = subscriptionId
-      if (!(queryName in this.subscriptions)) {
-        this.subscriptions[queryName] = []
+      if (!(subscriptionId in this.subscriptions)) {
+        this.subscriptions.set(subscriptionId, [])
       }
-      this.subscriptions[queryName].push(view)
+      this.subscriptions.get(subscriptionId).push(view)
       return subscriptionId
     },
     /**
      * Unsubscribe this view to a new GraphQL query.
-     * @param {string} queryName - Must be in QUERIES.
      * @param {number} subscriptionId - Subscription ID.
      */
-    unsubscribe (queryName, subscriptionId) {
-      if (queryName in this.subscriptions) {
+    unsubscribe (subscriptionId) {
+      if (this.subscriptions.has(subscriptionId)) {
         this.$workflowService.unsubscribe(subscriptionId)
+        this.subscriptions.delete(subscriptionId)
```

Comment:
Looks good, though views may request multiple subscriptions.

Reply:
Are you suggesting a change here?